deployer/src/args.rs

```diff
@@ -70,4 +70,8 @@ pub struct Args {
     /// Add an auth layer to deployer for local development
     #[arg(long)]
     pub local: bool,
+
+    // Posthog client key
+    #[clap(long)]
+    pub posthog_key: String,
```

Comment:
We'll need to add this to the makefile too, as well as adding the key for both staging and production to aws secrets manager + circleci before release. We then need to pass it in when we create the deployer config in gateway: https://github.com/shuttle-hq/shuttle/blob/main/gateway/src/project.rs#L872-L913. To know more about how this is done from makefile to gateway, I would look at how some of the other deployer args are handled.

Reply:
Make sure that any use of the key is optional (wherever we move this to), and that nothing happens if it is unset